### ISSUE TITLE
nixos/taskserver: Firewall is opened on all interfaces

### ISSUE THREAD

**Issue by tilpner** [NixOS/nixpkgs#68839]
**Describe the bug**

If `services.taskserver.listenHost` is changed from its default value of `localhost`, the module sets

```
networking.firewall.allowedTCPPorts = [ cfg.listenPort ];
```

This option is interface-agnostic, and doesn't allow the user to restrict access to a single
interface easily.

**Expected behavior**

There should be a way to allow access on only some interfaces, even if `listenHost` was set.

Some modules have an `openFirewall` option, which would solve this nicely. With a default of true, backwards compatibility would be kept.

I'll send a PR if that sounds good.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: services.taskserver.listenHost
```

cc @aszlig

**Comment by aszlig** [comment]
@tilpner: In addition to that option, I'd probably go for deprecation, removing the `allowedTCPPorts` definition entirely from the module and not open up those ports at all. The latter should be explicitly set by the user.

**Comment by tilpner** [comment]
Well, I never fixed it, so it's still a problem. :(

**Comment by liketechnik** [comment]
This seems to be [fixed on 22.05](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/nixos/modules/services/misc/taskserver/default.nix#L567=) after all.

**Comment by tilpner** [comment]
Indeed, fixed by 0091e3198a81cfc5cd867f676f3711a63979b938
